app/backend/core/authentication.py

```diff
@@ -0,0 +1,164 @@
+        return {
+            "useLogin": AuthenticationHelper.use_authentication(), # Whether or not login elements are enabled on the UI
+            "msalConfig": {
+                "auth": {
+                    "clientId": os.getenv("AZURE_CLIENT_APP_ID"), # Client app id used for login
+                    "authority": AuthenticationHelper._authority, # Directory to use for login https://learn.microsoft.com/azure/active-directory/develop/msal-client-application-configuration#authority
+                    "redirectUri": "/redirect", # Points to window.location.origin. You must register this URI on Azure Portal/App Registration.
+                    "postLogoutRedirectUri": "/", # Indicates the page to navigate after logout.
+                    "navigateToLoginRequestUrl": False # If "true", will navigate back to the original request location before processing the auth code response.
+                },
+                "cache": {
+                    "cacheLocation": "sessionStorage", # Configures cache location. "sessionStorage" is more secure, but "localStorage" gives you SSO between tabs.
+                    "storeAuthStateInCookie": False # Set this to "true" if you are having issues on IE11 or Edge
+                }
+            },
+            "loginRequest": {
+                # Scopes you add here will be prompted for user consent during sign-in.
+                # By default, MSAL.js will add OIDC scopes (openid, profile, email) to any login request.
+                # For more information about OIDC scopes, visit:
+                # https://docs.microsoft.com/azure/active-directory/develop/v2-permissions-and-consent#openid-connect-scopes
+                "scopes": ["api://{}/.default".format(os.getenv("AZURE_SERVER_APP_ID"))]
+            }
+        }
+
+    @staticmethod
+    def get_token_auth_header() -> str:
+        # Obtains the Access Token from the Authorization Header
+        auth = request.headers.get("Authorization", None)
+        if not auth:
+            raise AuthError({"code": "authorization_header_missing",
+                            "description":
+                            "Authorization header is expected"}, 401)
+
+        parts = auth.split()
+
+        if parts[0].lower() != "bearer":
+            raise AuthError({"code": "invalid_header",
+                            "description": "Authorization header must start with Bearer"}, 401)
+        elif len(parts) == 1:
+            raise AuthError({"code": "invalid_header",
+                            "description": "Token not found"}, 401)
+        elif len(parts) > 2:
+            raise AuthError({"code": "invalid_header",
+                            "description": "Authorization header must be Bearer token"}, 401)
+
+        token = parts[1]
+        return token
+
+    @staticmethod
+    def get_confidential_client():
+        # This example only uses the default memory token cache and should not be used for production
```

Comment:
It says this shouldnt be used for production?

Reply:
this was pasted in from the sample - i can use a token cache somewhere else

Reply:
found one in the msal-extensions library

Reply:
switched to a persisted encrypted token cache